src/core/column/cut.h

```diff
@@ -197,6 +196,87 @@ class Cut_ColumnImpl : public Virtual_ColumnImpl {
 };
 
 
+template <bool RIGHT_CLOSED>
+class CutBins_ColumnImpl : public Virtual_ColumnImpl {
+  private:
+    Column col_;
+    dblvec bins_;
+
+  public:
+
+    CutBins_ColumnImpl(Column&& col, dblvec&& bins)
+      : Virtual_ColumnImpl(col.nrows(), dt::SType::INT32),
+        col_(std::move(col)),
+        bins_(std::move(bins))
+    {
+      xassert(ltype_is_numeric(col_.ltype()));
+    }
+
+    ColumnImpl* clone() const override {
+      return new CutBins_ColumnImpl<RIGHT_CLOSED>(Column(col_), dblvec(bins_));
+    }
+
+    size_t n_children() const noexcept override {
+      return 1;
+    }
+
+    const Column& child(size_t i) const override {
+      xassert(i == 0);  (void)i;
+      return col_;
+    }
+
+    inline bool gt(double v1, double v2) const {
+      if (RIGHT_CLOSED) {
+        return v1 > v2;
+      } else {
+        return v1 >= v2;
+      }
+    }
+
+    inline bool lt(double v1, double v2) const {
+      if (RIGHT_CLOSED) {
+        return v1 <= v2;
+      } else {
+        return v1 < v2;
+      }
+    }
+
+
+    bool get_element(size_t i, int32_t* out) const override {
+      double value;
+      bool is_valid = col_.get_element(i, &value);
+
+      if (is_valid) {
+        size_t nbins = bins_.size();
+        is_valid = false;
+
+        if (gt(value, bins_[0]) && lt(value, bins_[nbins - 1])) {
+          is_valid = true;
+          *out = static_cast<int32_t>(bin_value(value, 0, nbins - 1));
```

Comment:
what if nbins==1? or 0?

Reply:
It cannot be, because this is indeed the number of the bin edges that we validate to be `>=2`. I probably need to invent a better name for this variable.